core/arch/arches.go

```diff
@@ -49,14 +48,70 @@ func (a Arches) String() string {
 	return strings.Join(a.set.SortedValues(), ",")
 }
 
-// ConstraintArch returns the arch for the constraint if there is one,
-// else it returns the default arch.
-func ConstraintArch(cons constraints.Value, defaultCons *constraints.Value) string {
-	if cons.HasArch() {
-		return *cons.Arch
+// The following constants define the machine architectures supported by Juju.
+const (
+	AMD64   = "amd64"
+	ARM64   = "arm64"
+	PPC64EL = "ppc64el"
+	S390X   = "s390x"
+	RISCV64 = "riscv64"
+)
+
+// AllSupportedArches records the machine architectures recognised by Juju.
+var AllSupportedArches = []string{
+	AMD64,
+	ARM64,
+	PPC64EL,
+	S390X,
+	RISCV64,
+}
+
+// UnsupportedArches records the machine architectures not supported by Juju.
+// Note: don't make const to prevent referencing it.
+var UnsupportedArches = []string{
+	"i386", "armhf",
+}
+
+// archREs maps regular expressions for matching
+// `uname -m` to architectures recognised by Juju.
+var archREs = []struct {
+	*regexp.Regexp
+	arch string
+}{
+	{Regexp: regexp.MustCompile("amd64|x86_64"), arch: AMD64},
+	{Regexp: regexp.MustCompile("aarch64"), arch: ARM64},
+	{Regexp: regexp.MustCompile("ppc64el|ppc64le"), arch: PPC64EL},
```

Comment:
Maybe add `ppc64` to the list.

From commit message: "Additionally, I've dropped support for ppc and we explicitly callout ppc64(le|el)." 
ppc64 was the listing in the arch package, which is still 64 bit, why drop it?